artifacts/reader.py

```diff
@@ -227,6 +196,49 @@ def _ReadSupportedOS(self, definition_values, definition_object, name):
 
     definition_object.supported_os = supported_os
 
+  def _ReadSources(self, artifact_definition_values, artifact_definition):
+    """Reads the artifact definition sources
+
+    Args:
+      artifact_definition_values: the artifact definition as a dict.
+      artifact_definition_values: the artifact definition object (instance of
+                           ArtifactDefinition).
+
+    Raises:
+      FormatError: if the type indicator is not set or unsupported,
+                   or if required attributes are missing.
+    """
+    name = artifact_definition_values.get(u'name', None)
+    sources = artifact_definition_values.get(u'sources')
+    if not sources:
+      raise errors.FormatError(
+          u'Invalid artifact definition: {0} missing sources.'.format(name))
```

Comment:
_ReadLabels does this too.  I can add the if block that raises if none, but in theory ReadArtifactDefinitionValues will raise if name is None before _ReadLabels or _ReadSources get called